crates/oxc_linter/src/lib.rs

```diff
@@ -82,8 +83,19 @@ impl Linter {
         fix: bool,
     ) -> Vec<Message<'a>> {
         let ctx = LintContext::new(source_text, semantic.clone(), fix);
+        let trivias = semantic.trivias();
+        let configuration_affected_lines = trivias.configuration_lines(source_text);
+        let nodes_to_lint = semantic
+            .nodes()
+            .iter()
+            .filter(|node| {
+                let span = node.get().kind().span();
+                let lines = trivias.line_numbers(source_text, span.start, span.end);
+                !configuration_affected_lines.iter().any(|l| lines.contains(l))
```

Comment:
If these two are ranges, then a simple inclusive range check should be enough?

Reply:
`configuration_affected_lines` is a `Vec<usize>`. The idea was to derive which lines are relevant, i.e. immediately following an `eslint-disable-next-line` for example